packages/flutter/test/cupertino/theme_test.dart

```diff
@@ -123,4 +135,24 @@ void main() {
       expect(theme.scaffoldBackgroundColor, CupertinoColors.black);
     },
   );
+
+  testWidgets('Theme has default IconThemeData, which is derived from the theme\'s primary color', (WidgetTester tester) async {
+      const Color primaryColor = CupertinoColors.destructiveRed;
+      const CupertinoThemeData themeData = CupertinoThemeData(primaryColor: primaryColor);
+
+      final IconThemeData resultingIconTheme = await testIconTheme(tester, themeData);
+
+      expect(resultingIconTheme.color, themeData.primaryColor);
+  });
+
+  testWidgets('IconTheme.of creates a dependency on iconTheme', (WidgetTester tester) async {
+      IconThemeData iconTheme = await testIconTheme(tester, const CupertinoThemeData(primaryColor: CupertinoColors.destructiveRed));
+
+      expect(buildCount, 1);
+      expect(iconTheme.color, CupertinoColors.destructiveRed);
+
+      iconTheme = await testIconTheme(tester, const CupertinoThemeData(primaryColor: CupertinoColors.activeOrange));
+      expect(buildCount, 2);
+      expect(iconTheme.color, CupertinoColors.activeOrange);
+  });
```

Comment:
Add a test in material to check that given a material theme with color A and a cupertino override theme with color B, the icons under that material theme still has color A.